src/import/sst_service.rs

```diff
@@ -341,12 +348,32 @@ impl<Router: RaftStoreRouter<RocksEngine>> ImportSst for ImportSSTService<Router
                     "output_level" => ?output_level, "err" => %e
                 ),
             }
-
-            future::result(res)
+            let res = engine.compact_files_in_range(start, end, output_level);
+            match res {
+                Ok(_) => info!(
+                    "compact files in range";
+                    "start" => start.map(log_wrappers::Key),
+                    "end" => end.map(log_wrappers::Key),
+                    "output_level" => ?output_level, "takes" => ?timer.elapsed()
+                ),
+                Err(ref e) => error!(
+                    "compact files in range failed";
+                    "start" => start.map(log_wrappers::Key),
+                    "end" => end.map(log_wrappers::Key),
+                    "output_level" => ?output_level, "err" => %e
+                ),
+            }
+            let res = res
                 .map_err(|e| Error::Engine(box_err!(e)))
-                .map(|_| CompactResponse::default())
-                .then(move |res| send_rpc_response!(res, sink, label, timer))
-        }))
+                .map(|_| CompactResponse::default());
+            send_rpc_response!(res, sink, label, timer);
+        };
+
+        let thread_handle = self.threads.clone();
+        let ctx_task = async move {
+            thread_handle.spawn(handle_task);
+        };
+        ctx.spawn(Compat::new(ctx_task.unit_error().boxed()));
```

Comment:
Could you explain what is happening here please? It's probably not wrong, but it seems odd to have two spawns for essentially one future.

Reply:
`ctx.spawn` represents the work done by the grpc thread. `thread_handle` represents the work done by the `ImportSSTService` thread pool. IMO spawn doesn't block current thread, it's ok to just spawn in current grpc thread not in `ctx.spawn`. What's your suggestion /cc @BusyJay

Reply:
Make sense.